labelbox/data/metrics/confusion_matrix/calculation.py

```diff
@@ -0,0 +1,288 @@
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+
+from ..iou.calculation import _get_mask_pairs, _get_vector_pairs, miou
+from ...annotation_types import (ObjectAnnotation, ClassificationAnnotation,
+                                 Mask, Geometry, Checklist, Radio,
+                                 ScalarMetricValue, ConfusionMatrixMetricValue)
+from ..group import (get_feature_pairs, get_identifying_key, has_no_annotations,
+                     has_no_matching_annotations)
+
+
+def confusion_matrix(ground_truths: List[Union[ObjectAnnotation,
+                                               ClassificationAnnotation]],
+                     predictions: List[Union[ObjectAnnotation,
+                                             ClassificationAnnotation]],
+                     include_subclasses: bool,
+                     iou: float) -> ConfusionMatrixMetricValue:
+    """
+    Computes the confusion matrix for an arbitrary set of ground truth and predicted annotations.
+    It first computes the confusion matrix for each metric and then sums across all classes
+
+    Args:
+        ground_truth : Label containing human annotations or annotations known to be correct
```

Comment:
i think we avoided using this name in the label import project because it didnt exist anywhere else in the application. Can you double check that we want to bake this into a public-facing method? I think maybe we can just call them `label`?

Reply:
Great point, we need to come up with some consistent language. The problem here is that Label means "data and annotations" not "human made annotations". There needs to be some distinction between human generated and machine generated labels. I am open to changing the name but I already exposed this concept with the iou metric functions for our first mea release :). Although these are positional args so if we want to change this we probably can. We should probably do it soon though.

Reply:
i honestly have no stake in it, i guess i just wanted to point it out. I think Gareth asked if we could change the name in label imports, so he might be a person to run it by. 
I definitely dont have a problem with it, and honestly, i think ground truth label is a good name, as thats the name in the industry, so i dont know why labelbox wouldnt go that direction.